compiler/rustc_hir/src/def.rs

```diff
@@ -190,6 +193,7 @@ impl DefKind {
             | DefKind::Variant
             | DefKind::Trait
             | DefKind::OpaqueTy
+            | DefKind::ImplTraitPlaceholder
```

Comment:
technically I think it's not really in a namespace at all since it doesn't have an ident, right?

Reply:
Sure, but the same thing could be said about `OpaqueTy`, except for possibly TAITs. I don't think this needs fixing now.